starter-code/routes/passportRouter.js

```diff
@@ -14,9 +14,60 @@ router.get("/private-page", ensureLogin.ensureLoggedIn(), (req, res) => {
   res.render("passport/private", { user: req.user });
 });
 
+router.get("/signup", (req, res, next) => {
+  res.render("passport/signup");
+});
+
+router.post("/signup", (req, res, next) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  if (username === "" || password === "") {
+    res.render("passport/signup", { message: "Indicate username and password" });
+    return;
+  }
+  User.findOne({ username }, "username", (err, user) => {
+    if (user !== null) {
+      res.render("passport/signup", { message: "The username already exists" });
+      return;
+    }
+
+    const salt = bcrypt.genSaltSync(bcryptSalt);
+    const hashPass = bcrypt.hashSync(password, salt);
+
+    const newUser = new User({
+      username,
+      password: hashPass
+    });
 
+    newUser.save((err) => {
+      if (err) {
+        res.render("passport/signup", { message: "Something went wrong" });
+      } else {
+        res.redirect("/");
+      }
+    });
+  });
+});
+
+// ---- LOGIN
 
+router.get("/login", (req, res, next) => {
+  
+  res.render("passport/login");
+});
 
+router.post("/login", passport.authenticate("local", {
+  successRedirect: "/private-page",
+  failureRedirect: "/login",
+  failureFlash: true,
```

Comment:
You are setting "failureFlash" to true but you are not passing any error message to the view (see previous comment).